src/v/cluster/data_migration_backend.cc

```diff
@@ -40,27 +54,753 @@ backend::backend(
+            _nodes_to_retry.try_emplace(node_id, new_deadline);
+            alarm = std::min(alarm, new_deadline);
+        }
+    }
+    _rpc_responses.clear();
+
+    // schedule fibers
+    for (auto node_id : to_send_rpc) {
+        _nodes_to_retry.erase(node_id);
+        co_await send_rpc(node_id);
+    }
+    spawn_advances();
+    if (alarm == model::timeout_clock::time_point::max()) {
+        _timer.cancel();
+    } else {
+        _timer.rearm(alarm);
+    }
+}
+
+void backend::wakeup() { _sem.signal(1 - _sem.available_units()); }
+
+[[nodiscard]] ss::future<std::optional<ssx::semaphore_units>> backend::lock() {
+    try {
+        co_return co_await _mutex.get_units(_as);
+    } catch (const ss::broken_semaphore&) {
+        co_return std::nullopt;
+    }
+}
+
+void backend::mark_rs_done_for_ntp(
+  migration_reconciliation_state& rs, const model::ntp& ntp) //
+{
+    auto& rs_topics = rs.outstanding_topics;
+    auto rs_topic_it = rs_topics.find({ntp.ns, ntp.tp.topic});
+    if (rs_topic_it != rs_topics.end()) {
+        auto& rs_parts = rs_topic_it->second.outstanding_partitions;
+        auto rs_part_it = rs_parts.find(ntp.tp.partition);
+        if (rs_part_it != rs_parts.end()) {
+            for (const auto& affected_node_id : rs_part_it->second) {
+                auto nstate_it = _nstates.find(affected_node_id);
+                nstate_it->second.erase(ntp);
+                if (nstate_it->second.empty()) {
+                    _nstates.erase(nstate_it);
+                    _nodes_to_retry.erase(affected_node_id);
+                }
+            }
+            rs_parts.erase(rs_part_it);
+            if (rs_parts.empty()) {
+                rs_topics.erase(rs_topic_it);
+            }
+        }
+    }
+}
+
+ss::future<> backend::send_rpc(model::node_id node_id) {
+    check_ntp_states_request req;
+    co_await ssx::async_for_each(
+      _nstates[node_id], [this, &req](const auto& pair) {
+          auto& [ntp, migration_id] = pair;
+          req.sought_states.push_back(
+            {.ntp = ntp,
+             .migration = migration_id,
+             .state = _mrstates.find(migration_id)->second.sought_state});
+      });
+
+    ssx::spawn_with_gate(
+      _gate, [this, node_id, req = std::move(req)]() mutable {
+          vlog(dm_log.debug, "sending RPC to node {}: {}", node_id, req);
+          ss::future<check_ntp_states_reply> reply
+            = (_self == node_id) ? check_ntp_states_locally(std::move(req))
+                                 : _frontend.check_ntp_states_on_foreign_node(
+                                   node_id, std::move(req));
+          return reply.then([node_id, this](check_ntp_states_reply&& reply) {
+              vlog(
+                dm_log.debug, "got RPC response from {}: {}", node_id, reply);
+              _rpc_responses[node_id] = std::move(reply);
+              return wakeup();
+          });
+      });
+}
+
+void backend::to_advance(id migration_id, state sought_state) //
```

Comment:
nit: why those comments `//` are needed, they seems to break the formatting we use across the whole code base

Reply:
I'd rather call it "fix", not "break". I find severely wrapped code hard to read, especially when an opening curly brace is not on a line indented the same as the matching closing one: it takes some mental capacity to separate function signature from body in case of a function, or condition from the statements in case of a conditional operator. If these violate our code style I can remove them.

Reply:
i guess it is a matter of preference, i have no personal preference. I asked about this as it would be great to have the code formatted in a consistent way across the tree. Long time ago we decided to based our code style on a google style which has inline curly braces.

Reply:
I guess the key difference is that google has 2 spaces for default indentations and 4 for wrapped lines, while we do exactly the opposite. I hope to get used to it sooner rather than later.